temporalcli/commandsgen/docs.go

```diff
@@ -75,27 +76,38 @@ func (w *docWriter) writeCommand(c *Command) {
 }
 
 func (w *docWriter) writeSubcommand(c *Command) {
-	fileName := strings.Split(c.FullName, " ")[1]
-	subCommand := strings.Join(strings.Split(c.FullName, " ")[2:], "")
-	w.fileMap[fileName].WriteString("## " + subCommand + "\n\n")
+	fileName := c.FileName
+	prefix := strings.Repeat("#", c.Depth)
+	w.fileMap[fileName].WriteString(prefix + " " + c.LeafName + "\n\n")
 	w.fileMap[fileName].WriteString(c.Description + "\n\n")
-	w.fileMap[fileName].WriteString("Use the following options to change the behavior of this command.\n\n")
-
-	// gather options from command and all options aviailable from parent commands
-	var allOptions = make([]Option, 0)
-	for _, options := range w.optionsStack {
-		allOptions = append(allOptions, options...)
-	}
 
-	// alphabetize options
-	sort.Slice(allOptions, func(i, j int) bool {
-		return allOptions[i].Name < allOptions[j].Name
-	})
+	if isLeafCommand(c) {
+		w.fileMap[fileName].WriteString("Use the following options to change the behavior of this command.\n\n")
 
-	for _, option := range allOptions {
-		w.fileMap[fileName].WriteString(fmt.Sprintf("## %s\n\n", option.Name))
-		w.fileMap[fileName].WriteString(option.Description + "\n\n")
+		// gather options from command and all options aviailable from parent commands
```

Comment:
```suggestion
		// gather options from command and all options available from parent commands
```

Also, an optional bit of feedback that can be improved upon later if you don't have time now:

I think I would try to keep the grouping similar to what we have in the actual CLI help output—global options like `--env` probably should be separate from command-specific options so there's less for users to sort through.

(I would ideally like to break this down even further along the lines of option sets, even in the `--help` output itself, but that's for another time.)

Reply:
Agree, would do that as a follow on.